packages/mui-material/src/styles/createThemeWithVars.js

```diff
@@ -68,7 +68,7 @@ function getSpacingVal(spacingInput) {
     return spacingInput;
   }
   if (typeof spacingInput === 'function') {
-    return getSpacingVal(spacingInput(1));
+    return spacingInput;
```

Comment:
Nit: Should we refactor to?:

```js
function getSpacingVal(spacingInput) {
  if (typeof spacingInput === 'number') {
    return `${spacingInput}px`;
  }
  if (Array.isArray(spacingInput) || ['string', 'function'].includes(typeof spacingInput)) {
    return spacingInput;
  }
  return '8px';
}
```

Reply:
Sure, but I am going to use `||` directly without `includes` because it's faster.